gallery_dl/extractor/instagram.py

```diff
@@ -140,6 +142,65 @@ def _extract_postpage(self, url):
 
         return medias
 
+    def _extract_stories(self, url):
+        page = self.request(url).text
+        shared_data = self._extract_shared_data(page)
+
+        # If no stories are present the URL redirect to `ProfilePage'
+        if 'StoriesPage' not in shared_data['entry_data']:
+            return []
+
+        user_id = shared_data['entry_data']['StoriesPage'][0]['user']['id']
+        variables = '{' + \
+            '"reel_ids":["{}"],'.format(user_id) + \
+            '"tag_names":[],"location_ids":[],' + \
+            '"highlight_reel_ids":[],"precomposed_overlay":true,' + \
+            '"show_story_viewer_list":true,' + \
+            '"story_viewer_fetch_count":50,"story_viewer_cursor":"",' + \
+            '"stories_video_dash_manifest":false}'
+        headers = {
+            "X-Requested-With": "XMLHttpRequest",
+        }
+        url = '{}/graphql/query/?query_hash={}&variables={}'.format(
+            self.root,
+            'cda12de4f7fd3719c0569ce03589f4c4',
+            variables,
+        )
+        shared_data = self.request(url, headers=headers).json()
+
+        # If there are stories present but the user is not authenticated or
+        # does not have permissions no stories are returned.
+        if not shared_data['data']['reels_media']:
+            return []   # no stories present
```

Comment:
Maybe add a warning as to why there aren't any stories?

Reply:
Good question!  What other extractors are doing in a similar situations?  Actually by using the web browser what happens is that no stories could be seen and a «Log in to view this story» form is showed.

Reply:
Hmm, now that you mention it: most other extractors also don't show any warnings etc. in such situations, even though they should. I guess it's OK to leave it like that for right now, and I'll go and add some more "authentication required" error messages to all other extractors.